packages/datastore/src/datastore.ts

```diff
@@ -34,13 +42,35 @@ import {
  * https://hal.inria.fr/file/index/docid/555588/filename/techreport.pdf
  */
 export
-class Datastore implements IIterable<Table> {
+class Datastore implements IIterable<Table<Schema>> {
+
+  /**
+   * Create a new datastore.
+   *
+   * @param id - The unique id of the datastore.
+   * @param schemas - The table schemas of the datastore.
+   */
+  constructor(id: number, schemas: ReadonlyArray<Schema>) {
+    this._context = {
+      inTransaction: false,
+      version: 0,
+      storeId: id,
+      change: {},
+      patch: {},
+    };
+    this._tables = {};
```

Comment:
You could use the new `BPlusTree` collection class to store the tables. It should be more space efficient and have faster iteration. Also, the complexity quotes in the comments are based on that class.